File: lib-connector/src/main/java/com/uis/connector/cache/Cache.java
package com.uis.connector.cache;

import java.io.File;
/**
 * @author uis
 */
public interface Cache {
    int NO_TIME_OUT = -1;
    String NO_PREFIX = "";
    /**
     * 获取内存缓存
     * @param key
     * @return json
     */
    String get(String key);

    /**
     * 获取内存缓存
     * @param key
     * @param mills 缓存时效（毫秒）
     * @return json
     */
    String get(String key, long mills);

    /**
     * 写入内存缓存
     * @param key
     * @param entity
     */
    void put(String key, Object entity);

    void put(String key, Object entity,long mills);

    /**
     * 写入磁盘缓存
     * @param key
     * @param value
     * @return
     */
    boolean writeFile(String key, Object value);

    /**
     * 读取磁盘缓存
     * @param key
     * @return json
     */
    String readFile(String key);

    /**
     * 读取磁盘缓存
     * @param key
     * @param mills 缓存时效（毫秒）
     * @return json
     */
    String readFile(String key, long mills);

    String readFile(String key, long mills,boolean saveCache);

    /**
     * 清除内存缓存
     * @param key
     */
    void clearMemory(String key);

    /**
     * 清除prefix前缀内存缓存
     */
    void clearAllMemory(String prefix);

    /**
     * 清除内存和文件缓存
     * @param key
     */
    void clear(String key);

    /**
     * 清除prefix前缀内存和文件缓存
     */
    void clearAll(String prefix);

    /**
     * 删除文件
     * @param path 文件路径
     */
    void deleteFile(String path);

    /**
     * 复制文件
     * @param res 源文件
     * @param des 目标文件
     * @return
     */
    boolean copyFile(File res,File des);

    /**
     * 保存到文件
     * @param isAppend true追加模式，false覆盖模式
     * @param data 数据
     * @param file 要保存的文件
     */
    void saveFile(boolean isAppend,byte[]data,File file);

    byte[] getFile(File file);
}

File: assignor/src/main/java/com/uis/assignor/cache/Cache.kt
/*
 * Copyright (c) 2019 by uis
 * Author: uis
 * Github: https://github.com/luiing
 */

package com.uis.assignor.cache

import java.io.File

const val NO_TIME_OUT = -1
const val NO_PREFIX = ""
const val MAX_CACHE_SIZE = 200

interface Cache {
    fun getCache(key: String, mills: Long = -1): String
    fun putCache(key: String, entity: Any?, mills: Long = System.currentTimeMillis())
    fun writeFile(key: String, value: Any?): Boolean
    fun readFile(key: String, mills: Long, saveCache: Boolean = false): String

    fun clearCache(key: String,isDisk :Boolean= false)
    fun clearGroupCache(prefix: String,isDisk :Boolean= false)

    fun deleteFile(path: String)
    fun copyFile(res: File, des: File): Boolean
    fun saveFile(data: ByteArray, file: File,append: Boolean = false )
    fun getFileBytes(file: File): ByteArray
}
